src/printer.rs

```diff
@@ -423,14 +428,16 @@ impl Printer {
 
     pub fn print_response_body(
         &mut self,
-        mut response: Response,
+        response: &mut Response,
         encoding: Option<&'static Encoding>,
         mime: Option<&str>,
     ) -> anyhow::Result<()> {
         let url = response.url().clone();
         let content_type =
             mime.map_or_else(|| get_content_type(response.headers()), ContentType::from);
-        let encoding = encoding.or_else(|| get_charset(&response));
+        let encoding = encoding.or_else(|| get_charset(response));
+        let compression_type = get_compression_type(response.headers());
+        let mut body = decompress_stream(response, compression_type);
```

Comment:
I think I did notice the change in the error message, but I can't remember the steps to reproduce it.

Reply:
Yep, with [flask](https://flask.palletsprojects.com/en/2.1.x/quickstart/#a-minimal-application):

```python
from flask import Flask

app = Flask(__name__)

@app.route('/badgz')
def badgz():
    return 'bad', {'Content-Encoding': 'gzip'}
```

Reply:
Thanks. So the piece I was missing is that small responses lead to `unexpected end of file`  while anything else will return the underlying error i.e `corrupt deflate stream` or `invalid gzip header`.

Slightly related, I was thinking of improving decompression error messages by modifying `decompress()` to return a Decompress/Decode wrapper that implements Read and prepends errors with `error decoding response body`. However, I think only io errors can be returned from the Read trait's methods so that is a dead end.

Reply:
You can [add a custom message](https://doc.rust-lang.org/std/io/struct.Error.html#method.new) to an `io::Error`.

Reply:
Nice, I will try using that now.